netx.go

```diff
@@ -5,27 +5,75 @@ package netx
 import (
 	"context"
 	"net"
+	"strings"
 	"sync/atomic"
 	"time"
 )
 
 var (
-	dial           atomic.Value
-	dialUDP        atomic.Value
-	listenUDP      atomic.Value
-	resolveTCPAddr atomic.Value
-	resolveUDPAddr atomic.Value
-
+	dial               atomic.Value
+	dialUDP            atomic.Value
+	listenUDP          atomic.Value
+	resolveTCPAddr     atomic.Value
+	resolveUDPAddr     atomic.Value
+	NAT64Prefix        atomic.Value
 	defaultDialTimeout = 1 * time.Minute
 )
 
 func init() {
 	Reset()
 }
 
+type NAT64PrefixHolder struct {
+	expiration time.Time
+	prefix     []byte
+}
+
+func getNAT64Prefix() []byte {
+	if holder, ok := NAT64Prefix.Load().(*NAT64PrefixHolder); holder != nil && ok {
+		if time.Now().Before(holder.expiration) {
```

Comment:
We probably need some way to expire this immediately when the ip address of the client changes? eg a wake on a new network or change of network due to movement or explicit network change.

Reply:
@forkner i've added a public method for resetting the prefix, just need to decide when/where to call it